src/frontends/pytorch/src/op/repeat_interleave.cpp

```
@@ -0,0 +1,93 @@
+    auto const_1 = context.mark_node(opset10::Constant::create(element::i32, Shape{}, {1}));
+    auto const_1_list = context.mark_node(opset10::Constant::create(element::i32, Shape{1}, {1}));
+    auto const_neg_1 = context.mark_node(opset10::Constant::create(element::i32, Shape{1}, {-1}));
+
+    // inputs
+    auto input = context.get_input(0);
+    std::shared_ptr<ov::Node> result;
+
+    auto repeats_ext_node = context.get_input_from_visible_context(1).get_node_shared_ptr();
+    auto repeats_fw_node = std::dynamic_pointer_cast<opset10::Constant>(repeats_ext_node);
+    if (repeats_fw_node && repeats_fw_node->cast_vector<int32_t>().size() > 1) {
+        // repeats is Constant with more then 1 element
+        auto repeats = repeats_fw_node->cast_vector<int32_t>();
+        if (context.input_is_none(2)) {
+            // case (repeats=tensor, dim=None)
+            auto flat_shape = context.mark_node(opset10::Constant::create(element::i32, Shape{1}, {-1}));
+            auto reshape = context.mark_node(std::make_shared<opset10::Reshape>(input, flat_shape, false));
+            OutputVector all_indices = generate_indices_from_repeats_tensor(repeats, context);
+            auto concat = context.mark_node(std::make_shared<opset10::Concat>(all_indices, 0));
+            result = std::make_shared<opset10::Gather>(reshape, concat, const_0);
+        } else {
+            // case (repeats=tensor, dim=number)
+            auto dimension = context.get_input(2);
+            OutputVector all_indices = generate_indices_from_repeats_tensor(repeats, context);
+            auto concat = context.mark_node(std::make_shared<opset10::Concat>(all_indices, 0));
+            result = std::make_shared<opset10::Gather>(input, concat, dimension);
+        }
+    } else {
+        // repeats is not Constant or single element constant
+        // Curently we support only case when repeats contains only one element. Otherwise next Reshape will fail.
```

I've checked how it fails, unfortunately it seem to just raise runtime error with exception from Reshape instead of just returning message of unconverted operators. Since this is known limitation, I think better exception handling would be needed

---

Actually no exception here is thrown, because on the conversion stage shape of `repeats` is not known. Exception is thrown during "infer shapes and types" stage.

---

My bad, I though that failure was during conversion, not infer_shapes. In that case, I don't think we can do much more